packages/app/metrics-utils/src/AbstractCounterMetric.ts

```diff
@@ -0,0 +1,87 @@
+import type { RequestContext } from '@lokalise/fastify-extras'
+import type { IFastifyMetrics } from 'fastify-metrics'
+import type { Counter } from 'prom-client'
+
+type CounterMetricConfiguration<
+  TMetricLabel extends string,
+  TMetricMeasurementKeys extends string[],
+> = {
+  name: string
+  helpDescription: string
+  label: TMetricLabel
+  measurementKeys: TMetricMeasurementKeys
+}
+
+export abstract class AbstractCounterMetric<
+  TMetricLabel extends string,
+  TMetricMeasurementKeys extends string[],
+> {
+  private readonly counterMetric: Counter<TMetricLabel> | null
```

Comment:
can you add comment explaining it's null when metrics are disabled on app level?

Reply:
Comment added ✅